examples/parallel_memmap.py

```diff
@@ -23,43 +99,75 @@ def sum_row(input, output, i):
     output[i] = sum_
 
 
+###############################################################################
+# We create a large 2D array containing some data.
+
+import numpy as np
+
 rng = np.random.RandomState(42)
-folder = tempfile.mkdtemp()
+samples = rng.normal(size=(10, int(1e6)))
+
+###############################################################################
+# Prepare the folder where the memmap will be dumped.
+
+import os
+
+folder = './joblib_memmap'
+try:
+    os.mkdir(folder)
+except FileExistsError:
+    pass
 samples_name = os.path.join(folder, 'samples')
 sums_name = os.path.join(folder, 'sums')
+
+###############################################################################
+# Pre-allocate a writeable shared memory map as a container for the results of
+# the parallel computation and dump the samples buffer.
+
+from joblib import dump
+
+sums = np.memmap(sums_name, dtype=samples.dtype,
+                 shape=samples.shape[0], mode='w+')
+dump(samples, samples_name)
+
+###############################################################################
+# Release the reference on the original in memory array and replace it by a
+# reference to the memmap array so that the garbage collector can release the
+# memory before forking. gc.collect() is internally called in Parallel just
+# before forking.
```

Comment:
You don't need to mention that gc.collect()... Most of our users will not understand what garbage collection is.

I think that this whole paragraph can be written as:

"We replace the 'samples' array by its memory mapped version".

We don't need to talk about garbage collection.